### ISSUE TITLE
Creating a Fortran module from the YAML dictionary file

### ISSUE THREAD

**Issue by rmontuoro** [ESCOMP/PhysicalConstantsDictionary#8]
It is generally agreed that physical constants should be included in models at build time as parameter sets provided by a Fortran module. The Fortran module file will be generated by parsing the physical constant dictionary YAML file.

A suitable mapping method needs to be established from YAML data hierarchies to Fortran's flat parameter lists. This method could leverage a well-defined syntax for the Fortran parameter names.

This issue is related to #2

**Comment by gold2718** [comment]
I feel that the Fortran module should only have one instance of any constant. When multiple values are available (via different sets) in the dictionary, selecting one at Fortran module generation time (via an option to the script) allows models to use a single symbol in their use statements.

**Comment by apcraig** [comment]
I agree with @gold2718.  When creating a Fortran module, the constants defined should already be constrained by a defined "set".

Whether that Fortran module should contain all constants possible or just those associated with the set is a separate question.  But I would lean toward some flexibility whereby a user could define a set of sets with some hierarchy defined by the order of the list.

Lets give a simple example.  Say you had

  earth_gravity <set=default,model1> = 9.8
  earth_radius <set=default> = 6371.
  earth_radius <set=model1> = 6370.
  mars_radius <set=default> = 3389.
  number_of_birds <set=birdmodel> = 5120

I would say the following.  If you asked for a Fortran file for defaults, it would produce

  earth_gravity = 9.8
  earth_radius = 6371
  mars_radius = 3389

If you asked for a Fortran file for model1, it would produce just

  earth_gravity = 9.8
  earth_radius = 6370

and if you asked for a Fortran file for default+model1, it would produce

  earth_gravity = 9.8
  earth_radius = 6370
  mars_radius = 3389

You would not be restricted to just asking for a set of constants for a single or pair of sets, but as many as you like (default+model1+mylocalmods for instance).  The yaml file would then provide flexibility for a community to create a limited, well-defined set of constants OR to create a set of constants that are slightly modified from another set of constants OR to write out all possible constants based on some hierarchy of sets.  All can be useful.

I don't know whether "default" is something that is being consider at this point.  I sense this is all leaning toward published constants, but models may use a mix of those published constants or their own.  So while published constants is one group of sets, model-centric sets will also be required that are not published.  And each modeling community will have to decide how to extract their set from the yaml file.

One final point, how do you secure the implementation so when the file is edited, it doesn't break other communities?
